README.md

```diff
@@ -115,3 +115,40 @@ using AutoViz
 set_color_theme(LIGHTTHEME)
 ```
 You can also define your own color theme using a dictionary. Look at the example in `src/colorscheme.jl` to have the correct key names.
+
+
+## Change Log
+
+### v0.8.x
+
+#### Rendering
+ - Clean-up of the rendering interface: there is now only one single rendering function with the signature
```

Comment:
Does this mean the user has to construct a `RenderModel` object explicitly?

Reply:
When you say "there is now only one single rendering function" do you mean there is only one function for people to implement to make their type renderable, or do you mean that there is only one function that a user can call if they want to render a scene?

Reply:
Yes, I think the typical usage now is as follows:
```
rm = RenderModel()
update_camera!(rm, MyCamera()) 
render!(rm, [my_renderables...]) 
```

I think it would still be convenient to have **one** `render` function that allows to do things in one function call and have the explicit version if one wants more fine tuning.

Reply:
Yes, `RenderModel` needs to be constructed explicitly. This is mostly because `RenderModel` holds the main camera parameters (`camera_center`, `camera_zoom`, `camera_rotation`). This is not by my choice but rather how `RenderModel` was designed to work. In fact if designing from scratch I would probably not have the `RenderModel` hold those camera parameters.
So in order to set the initial zoom level (for example), it needs to be passed to the `RenderModel`. In the previous version, the zoom level was set by first setting the zoom level of a camera and then the camera would go set the zoom level of the rendermodel, but effectively those variables were duplicated in `RenderModel` and `Camera` which I was not a big fan of. Would you agree?

Reply:
On a conceptual level, the camera state really lives inside the `RenderModel` (again, based on legacy code) and the structs that are called `Camera` actually describe camera motions over time (like "stay static", "follow a vehicle", "follow the scene"). This is quite subtle but I think the struct names may be a little misleading.

Reply:
Probably worth providing one or two convenience functions that take care of constructing a `RenderModel` but for full control I think we need all three steps (`RenderModel`, `update_camera!`, `render!`)

Reply:
@MaximeBouton added an "easy" render function that just takes a scene and optional overlays. There are still a few open questions there regarding the defaults

Reply:
I will try it out today. 

What would be your suggestion on how to handle the camera if we did not have the `RenderModel` struct?

Reply:
for the default render method this is what I had in mind: 
```
"""
Takes care of initializing a `RenderModel` and updating the camera.
For full control, use `render!(rendermodel, renderables)` instead.
"""
function render(
    renderables;
    camera_zoom::Float64 = 10.,
    camera_center::VecE2 = VecE2(0., 0.),
    camera_rotation::Float64 = 0.,
    camera_motion::Camera = SceneFollowCamera(),
    canvas_width::Int=DEFAULT_CANVAS_WIDTH,
    canvas_height::Int=DEFAULT_CANVAS_HEIGHT,
    surface::CairoSurface = CairoSVGSurface(IOBuffer(), canvas_width, canvas_height),
)
    rendermodel = RenderModel(camera_center=camera_center, 
                              camera_zoom=camera_zoom, 
                              camera_rotation=camera_rotation)
    update_camera!(rendermodel, camera_motion)
    render!(rendermodel, renderables)

    return surface
end
```

no need to go back to something specific that takes `scene` and `roadway` as input.

Reply:
> I will try it out today.
> 
> What would be your suggestion on how to handle the camera if we did not have the `RenderModel` struct?

One Idea that crossed my mind would be to have a `CameraState` object that has the fields `zoom`, `position`, `rotation`. Each `Camera` then has a reference to a `CameraState` which it then manipulates. And when we pass in the `Camera` to the `render` function, with it comes the `CameraState` object. In addition to the `CameraState`, `Camera` itself would hold other variables that parameterize how it moves over the scene.
`RenderModel` would then only keep track of the rendering instructions essentially. In this way, maybe the canvas dimension and canvas surface too, didn't spend much thought on this.
In this way, we would not need to pass in the `RenderModel` but only the `Camera`.

Reply:
> for the default render method this is what I had in mind:
> 
> ```
> """
> Takes care of initializing a `RenderModel` and updating the camera.
> For full control, use `render!(rendermodel, renderables)` instead.
> """
> function render(
>     renderables;
>     camera_zoom::Float64 = 10.,
>     camera_center::VecE2 = VecE2(0., 0.),
>     camera_rotation::Float64 = 0.,
>     camera_motion::Camera = SceneFollowCamera(),
>     canvas_width::Int=DEFAULT_CANVAS_WIDTH,
>     canvas_height::Int=DEFAULT_CANVAS_HEIGHT,
>     surface::CairoSurface = CairoSVGSurface(IOBuffer(), canvas_width, canvas_height),
> )
>     rendermodel = RenderModel(camera_center=camera_center, 
>                               camera_zoom=camera_zoom, 
>                               camera_rotation=camera_rotation)
>     update_camera!(rendermodel, camera_motion)
>     render!(rendermodel, renderables)
> 
>     return surface
> end
> ```
> 
> no need to go back to something specific that takes `scene` and `roadway` as input.

I wish this were possible, it was also what I initially started off with. But I currently don't see an easy way of doing this:
 - `SceneFollowCamera` and `TargetFollowCamera` both need access to the scene in order to determine their new position (this is why we had issue #9 where `SceneFollowCamera` would not work in the new rendering interface). The best we could do would be to default to `StaticCamera` but not sure if the convenience is worth what we're giving up.
 - We need to know what is the roadway so that we can render it before (i.e. below) the scene. If we pass in the roadway with the other renderables, we would have to search through the renderables and identify the roadway by type and render it first, which is pretty ugly.
The only way i see around that would be to give every renderable object a depth parameter which we can query using `depth(obj)` and then determine rendering order based on that. Still not really the way to go I think :/ 
So given these two constraints, this "convenience" render function was the best I could think of...